main.tf

```diff
@@ -1,10 +1,179 @@
+    },
+    {
+      name  = "ondat-operator.cluster.create"
+      value = var.create_cluster
+    },
+    {
+      name  = "ondat-operator.cluster.secretRefName"
+      value = "storageos-api"
+    },
+    {
+      name  = "ondat-operator.cluster.kvBackend.address"
+      value = local.ondat_etcd_endpoints
+    },
+    {
+      name  = "ondat-operator.cluster.kvBackend.tlsSecretName"
+      value = length(kubernetes_secret.etcd) > 0 ? kubernetes_secret.etcd[0].metadata[0].name : "storageos-etcd-secret"
+    },
+    {
+      name  = "ondat-operator.cluster.kvBackend.tlsSecretNamespace"
+      value = length(kubernetes_secret.etcd) > 0 ? kubernetes_secret.etcd[0].metadata[0].namespace : "storageos"
+    },
+    {
+      name  = "etcd-cluster-operator.cluster.create"
+      value = length(var.etcd_endpoints) == 0
+    },
+  ]
+
+  set_sensitive_values = [
+    {
+      name  = "cluster.admin.username",
+      value = var.admin_username
+    },
+    {
+      name  = "cluster.admin.password",
+      value = var.admin_password
+    },
+  ]
+
+  irsa_config = {
+    create_kubernetes_namespace = false
+    kubernetes_namespace        = kubernetes_namespace.ondat.metadata[0].name
+
+    create_kubernetes_service_account = true
+    kubernetes_service_account        = local.service_account_name
+
+    iam_role_path                 = "/"
+    tags                          = var.addon_context.tags
+    eks_cluster_id                = var.addon_context.eks_cluster_id
+    irsa_iam_policies             = var.irsa_policies
+    irsa_iam_permissions_boundary = var.irsa_permissions_boundary
+  }
+
+  addon_context = var.addon_context
+}
+
+resource "kubernetes_namespace" "ondat" {
+  metadata {
+    name = "ondat"
+    labels = {
+      app = local.name
+    }
+  }
+}
+
+################################################################################
+# Secrets
+################################################################################
+
+resource "kubernetes_namespace" "storageos" {
+  count = length(var.etcd_endpoints) == 0 ? 0 : 1
+
+  metadata {
+    name = "storageos"
+    labels = {
+      app = local.name
+    }
+  }
+}
+
+resource "kubernetes_secret" "etcd" {
+  count = length(var.etcd_endpoints) == 0 ? 0 : 1
+
+  metadata {
+    name      = "storageos-etcd"
+    namespace = kubernetes_namespace.storageos[0].metadata[0].name
+    labels = {
+      app = local.name
+    }
+  }
+
+  data = {
+    "etcd-client-ca.crt" = var.etcd_ca
+    "etcd-client.crt"    = var.etcd_cert
+    "etcd-client.key"    = var.etcd_key
+  }
+
+  type = "kubernetes.io/storageos"
+}
+
+################################################################################
+# Storage Class
+################################################################################
+
+resource "kubernetes_storage_class" "etcd" {
+  count = length(var.etcd_endpoints) == 0 ? 1 : 0
+
+  metadata {
+    name = "etcd"
+  }
+
+  storage_provisioner = "ebs.csi.aws.com"
```

Comment:
This seems like the EBS CSI driver is a hard requirement for Ondat, correct? Is that stated/shown anywhere?

Reply:
for the etcd backing store - yes its a requirement, `gp3` is strongly recommended better performance > https://docs.ondat.io/docs/install/aws/amazon-web-services-eks/#5---creating-a-storageclass-for-etcd-to-use